File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/base/model/FingerprintManagerCancellationSignal.java
package com.jesusm.jfingerprintmanager.base.model;

import android.support.v4.os.CancellationSignal;

public class FingerprintManagerCancellationSignal {
    private CancellationSignal cancellationSignal;

    public CancellationSignal getCancellationSignal() {
        return cancellationSignal;
    }

    public void start() {
        cancel();
        cancellationSignal = new CancellationSignal();
    }

    public boolean isCancelled() {
        return cancellationSignal == null;
    }

    public void cancel() {
        if (!isCancelled()) {
            cancellationSignal.cancel();
            cancellationSignal = null;
        }
    }
}

File: kfingerprintmanager/src/main/kotlin/com/jesusm/kfingerprintmanager/base/model/FingerprintManagerCancellationSignal.kt
package com.jesusm.kfingerprintmanager.base.model

import android.support.v4.os.CancellationSignal

class FingerprintManagerCancellationSignal(val cancellationSignal: CancellationSignal = CancellationSignal()) {
    var isCancelled : Boolean = false

    fun start() {
        cancel()
        isCancelled = false
    }

    fun cancel() {
        if (isCancelled.not()) {
            cancellationSignal.cancel()
            isCancelled = true
        }
    }
}